fv3core/pace/fv3core/initialization/geos_wrapper.py

```diff
@@ -0,0 +1,256 @@
+        pk: np.ndarray,
+        peln: np.ndarray,
+        pkz: np.ndarray,
+        phis: np.ndarray,
+        q_con: np.ndarray,
+        omga: np.ndarray,
+        ua: np.ndarray,
+        va: np.ndarray,
+        uc: np.ndarray,
+        vc: np.ndarray,
+        mfxd: np.ndarray,
+        mfyd: np.ndarray,
+        cxd: np.ndarray,
+        cyd: np.ndarray,
+        diss_estd: np.ndarray,
+    ):
+        isc = self._grid_indexing.isc
+        jsc = self._grid_indexing.jsc
+        iec = self._grid_indexing.iec + 1
+        jec = self._grid_indexing.jec + 1
+
+        # Assign compute domain:
+        self.dycore_state.u.view[:] = u[isc:iec, jsc : jec + 1, :-1]
+        self.dycore_state.v.view[:] = v[isc : iec + 1, jsc:jec, :-1]
+        self.dycore_state.w.view[:] = w[isc:iec, jsc:jec, :-1]
+        self.dycore_state.ua.view[:] = ua[isc:iec, jsc:jec, :-1]
+        self.dycore_state.va.view[:] = va[isc:iec, jsc:jec, :-1]
+        self.dycore_state.uc.view[:] = uc[isc : iec + 1, jsc:jec, :-1]
+        self.dycore_state.vc.view[:] = vc[isc:iec, jsc : jec + 1, :-1]
+
+        self.dycore_state.delz.view[:] = delz[isc:iec, jsc:jec, :-1]
+        self.dycore_state.pt.view[:] = pt[isc:iec, jsc:jec, :-1]
+        self.dycore_state.delp.view[:] = delp[isc:iec, jsc:jec, :-1]
+
+        self.dycore_state.mfxd.view[:] = mfxd[isc : iec + 1, jsc:jec, :-1]
+        self.dycore_state.mfyd.view[:] = mfyd[isc:iec, jsc : jec + 1, :-1]
+        self.dycore_state.cxd.view[:] = cxd[isc : iec + 1, jsc:jec, :-1]
+        self.dycore_state.cyd.view[:] = cyd[isc:iec, jsc : jec + 1, :-1]
+
+        self.dycore_state.ps.view[:] = ps[isc:iec, jsc:jec]
+        self.dycore_state.pe.view[:] = pe[isc:iec, jsc:jec, :]
+        self.dycore_state.pk.view[:] = pk[isc:iec, jsc:jec, :]
+        self.dycore_state.peln.view[:] = peln[isc:iec, jsc:jec, :]
+        self.dycore_state.pkz.view[:] = pkz[isc:iec, jsc:jec, :-1]
+        self.dycore_state.phis.view[:] = phis[isc:iec, jsc:jec]
+        self.dycore_state.q_con.view[:] = q_con[isc:iec, jsc:jec, :-1]
+        self.dycore_state.omga.view[:] = omga[isc:iec, jsc:jec, :-1]
+        self.dycore_state.diss_estd.view[:] = diss_estd[isc:iec, jsc:jec, :-1]
+
+        # tracer quantities should be a 4d array in order:
+        # vapor, liquid, ice, rain, snow, graupel, cloud
+        self.dycore_state.qvapor.view[:] = q[isc:iec, jsc:jec, :-1, 0]
+        self.dycore_state.qliquid.view[:] = q[isc:iec, jsc:jec, :-1, 1]
+        self.dycore_state.qice.view[:] = q[isc:iec, jsc:jec, :-1, 2]
+        self.dycore_state.qrain.view[:] = q[isc:iec, jsc:jec, :-1, 3]
+        self.dycore_state.qsnow.view[:] = q[isc:iec, jsc:jec, :-1, 4]
+        self.dycore_state.qgraupel.view[:] = q[isc:iec, jsc:jec, :-1, 5]
+        if self.namelist["dycore_config"]["nwat"] > 6:
+            self.dycore_state.qcld.view[:] = q[isc:iec, jsc:jec, :-1, 6]
+
+    def _prep_outputs_for_geos(self):
+        self.output_dictionary["u"] = self.dycore_state.u.data[:]
```

Comment:
Could this return the dict instead of mutating self? Again, along the lines of exposing the data flow in the interface instead of mutation.